pgrx/src/xid.rs

```diff
@@ -36,3 +43,84 @@ fn convert_xid_common(xid: pg_sys::TransactionId, last_xid: u32, epoch: u32) ->
 
     (epoch << 32) | xid as u64
 }
+
+/// An `inet` type from PostgreSQL
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq)]
+pub struct Xid(pg_sys::TransactionId);
```

Comment:
Is this what we want?  

I'm not _super_ keen on wrapping such a fundamental internal Postgres type.  I feel like doing so implies there's some difference between this new `Xid` type and the Postgres internal `pg_sys::TransactionId` -- and there's not.

I'm assuming you did this mostly for the serde support?  `pg_sys::TransactionId` will serialize just fine as it's a plain `u32`.

Reply:
I would do that, but then any `u32` will be `xid`. Do you think that `CommandId` or any other alias of `u32` is `xid`? I don't.

Reply:
You are correct.  I had forgotten that type aliases are presented to our schema generator as the type they alias.

Then what we need is some handing around `pg_sys::TransactionId` in a manner similar to how we have our own `pg_sys::Datum` impl.  This way, the type is still called `pg_sys::TransactionId` internally, which is my primary requirement here, and we can add whatever From traits or otherwise we think we want directly in the `pgrx-pg-sys` crate.